alicloud/data_source_alicloud_mongo_instances.go

```diff
@@ -0,0 +1,269 @@
+package alicloud
+
+import (
+	"log"
+	"regexp"
+	"time"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/requests"
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/dds"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceAlicloudMongoInstances() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAlicloudMongoInstancesRead,
+
+		Schema: map[string]*schema.Schema{
+			"name_regex": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateNameRegex,
+			},
+			"engine": {
```

Comment:
If engine only has one validate value, the field can be removed and new flied 'availability_zone' and 'instance_class' can be added.